### PR TITLE
Add load certificate test and support non-standard Go versions in make

### PR DESCRIPTION
**Pull request by thom-at-redhat**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,8 +46,10 @@ else
 	DEBUGFLAGS=
 endif
 
+GO ?= go
```
Comment (AaronH88) [inline Makefile:49]: Any reason for this? 

If so, can you please add this reason to the PR :smile:

File: pkg/certificates/ca.go [hunk 2]
```diff
@@ -21,6 +21,16 @@ import (
 	"github.com/ansible/receptor/pkg/utils"
 )
 
+type Oser interface {
+	ReadFile(name string) ([]byte, error)
+}
+
+type OsWrapper struct{}
+
+func (ow *OsWrapper) ReadFile(name string) ([]byte, error) {
```
Comment (AaronH88) [inline pkg/certificates/ca.go:30]: very nice

File: Makefile (hunk 1 above)
Comment (thom-at-redhat) [reply Makefile:49]: Added, the makefile previously had tests failing with  go 1.20.  this added non-disruptive support for multiple  Go versions

**Review by AaronH88 — APPROVED** [review]
lgtm

**Pull request merged**